### PR TITLE
traffic steering container attributes

### PR DESCRIPTION
**Pull request by aloktiagi**
This change introduces the NISE (Network Interception and Steering Entity) Titus System Service

The service is responsible for intercepting all incoming and outgoing TCP connections on Titus container and delivering them to ProxyD

### CODE REVIEW

File: executor/runtime/types/pod.go [hunk 1]
```diff
@@ -587,6 +588,10 @@ func (c *PodContainer) ServiceMeshEnabled() bool {
 	return c.serviceMeshEnabled
 }
 
+func (c *PodContainer) TrafficSteeringEnabled() bool {
+	return c.trafficSteeringEnabled
```
Comment (solarkennedy) [inline executor/runtime/types/pod.go:592]: This is missing something. It is never set!

There needs to be some code somewhere in pod.go that looks for a special Annotation and sets this to be true.

This is the "new" style of attributes with pods (aka v1 pod schema)

File: executor/runtime/types/sidecars.go [hunk 2]
```diff
@@ -124,6 +125,12 @@ var sideCars = []ServiceOpts{
 			"/titus/container-tools": {},
 		},
 	},
+	{
+		ServiceName:  SidecarTrafficSteering,
+		UnitName:     "titus-sidecar-traffic-steering",
```
Comment (solarkennedy) [inline executor/runtime/types/sidecars.go:130]: I don't see this unit. 

Is this really a new systemd titus system service?

**Comment by solarkennedy** [comment]
> ### Description of the Change
> **FIXME**: please add at least one sentence explaining why changes here are being made.

Can you please fill this in with *some* context about your change??

**Comment by solarkennedy** [comment]
Please rebase against master to fix conflicts and buildkite fixes

File: executor/runtime/types/sidecars.go (hunk 2 above)
Comment (aloktiagi) [reply executor/runtime/types/sidecars.go:131]: This has been added

File: executor/runtime/docker/docker.go [hunk 3]
```diff
@@ -1054,6 +1054,14 @@ func (r *DockerRuntime) Prepare(ctx context.Context, pod *v1.Pod) error { // nol
 		})
 	}
 
+	if runtimeTypes.GetSidecarConfig(systemServices, runtimeTypes.SidecarTrafficSteering).EnabledCheck(&r.cfg, r.c) {
+		if r.c.TrafficSteeringEnabled() {
+			r.c.SetEnvs(map[string]string{
+				"TITUS_SECCOMP_AGENT_HANDLE_TRAFFIC_STEERING": "true",
```
Comment (solarkennedy) [inline executor/runtime/docker/docker.go:1060]: Are you sure this is a new Titus System Service?

This variable kinda sounds like it is handled by the existing TSA

**Review by solarkennedy — CHANGES_REQUESTED** [review]

File: executor/runtime/docker/docker.go (hunk 3 above)
Comment (aloktiagi) [reply executor/runtime/docker/docker.go:1060]: Yeah pretty sure its a new TSS :) The systemd file has a new binary 
`
ExecStart=/usr/bin/nise -task_id %i
`

https://docs.google.com/document/d/1E3KnpagrgQDh5rellBU1-hYWf5e6TbQ_z--y6wxN58I/edit#heading=h.n4vq1kaspkzl

TSA is used as well to make the whole solution work.

File: executor/runtime/types/container.go [hunk 4]
```diff
@@ -71,6 +71,7 @@ const (
 	subnetsParam                            = "titusParameter.agent.subnets"
 	elasticIPPoolParam                      = "titusParameter.agent.elasticIPPool"
 	elasticIPsParam                         = "titusParameter.agent.elasticIPs"
+	TrafficSteeringEnabledParam             = "titusParameter.agent.trafficSteeringEnabled"
```
Comment (solarkennedy) [inline executor/runtime/types/container.go:74]: This needs some equivalent code in pod.go

**Review by MyUmmaGumma — APPROVED** [review]

File: executor/runtime/types/pod.go [hunk 5]
```diff
@@ -614,6 +614,13 @@ func (c *PodContainer) ServiceMeshEnabled() bool {
 	return c.serviceMeshEnabled
 }
 
+func (c *PodContainer) TrafficSteeringEnabled() bool {
+	if c.podConfig.TrafficSteeringEnabled != nil {
```
Comment (solarkennedy) [inline executor/runtime/types/pod.go:618]: great

**Review by solarkennedy — APPROVED** [review]

**Comment by solarkennedy** [comment]
nice job getting this into shape

**Pull request merged**